roles/ceph-osd/tasks/docker/start_docker_osd.yml

```diff
@@ -15,25 +15,31 @@
   with_items: ceph_osd_docker_devices
   register: osd_prepared
 
+# use shell rather than docker module
+# to ensure osd disk prepare finishes before
+# starting the next task
 - name: prepare ceph osd disk
-  docker:
-    image: "{{ ceph_osd_docker_username }}/{{ ceph_osd_docker_imagename }}"
-    name: "{{ ansible_hostname }}-osd-prepare-{{ item.0 | regex_replace('/', '') }}"
-    net: host
-    pid: host
-    state: started
-    privileged: yes
-    env: "OSD_DEVICE={{ item.0 }},{{ ceph_osd_docker_prepare_env }}"
-    volumes: "/var/lib/ceph:/var/lib/ceph,/etc/ceph:/etc/ceph,/dev:/dev,/run:/run"
+  shell: |   
+    docker run -ti --net=host \
```

Comment:
the problem I see is that docker module doesn't watch container exit status, so the task may return before the command is finished (same with `docker -d`). Then when next task (osd activate) is started, the osd is still being prepared and that causes problem.